lib/metasploit/framework/tcp/client.rb

```diff
@@ -83,6 +83,7 @@ def connect(global = true, opts={})
 
           nsock = Rex::Socket::Tcp.create(
               'PeerHost'      =>  opts['RHOST'] || rhost,
+              'PeerHostname'  =>  opts['SSLServerNameIndication'] || opts['VHOST'] || opts['RHOSTNAME'],
```

Comment:
Seems like `RHOSTNAME` should be prioritized over `VHOST`. `VHOST` is specific to HTTP and under normal conditions would presumably be the same as `RHOSTNAME`. Since this is the TCP connection though, if `RHOSTNAME` and `VHOST` are different, I would expect `RHOSTNAME` to be for the connection and `VHOST` to be used for the HTTP header.

Reply:
The http modules don't go down this codepath; and I was a bit on the fence about adding this logic here at all to be honest. Any thoughts on removing it entirely?

```suggestion
              'PeerHostname'  =>  opts['SSLServerNameIndication'] || opts['RHOSTNAME'],
```

Reply:
Yeah I think that would make it more clear.